[New library] System UI Controller (#270)
google/accompanist: docs/systemuicontroller.md

# System UI Controller for Jetpack Compose

[![Maven Central](https://img.shields.io/maven-central/v/com.google.accompanist/accompanist-systemuicontroller)](https://search.maven.org/search?q=g:com.google.accompanist)

System UI Controller provides easy-to-use utilities for updating the System UI bar colors within Jetpack Compose.

## Usage

> To setup Insets in your composables, you need to call the provide an instance to the `LocalSystemUiController`

Reviewer: ```suggestion
To control system UI in your composables, you must provide an instance to the `LocalSystemUiController`
```

composition local. This would typically be done near the top level of your composable hierarchy:

``` kotlin
setContent {
    // Create a controller, and provide it to the LocalSystemUiController

>     val controller = androidSystemUiController(LocalView.current)

Reviewer: This should be remembered right? Or should we offer a `rememberAndroidSBC` method?

    CompositionLocalProvider(LocalSystemUiController provides controller) {
        MyHomeScreen()
    }
}
```

Then in your layouts, you can update the system bar colors as necessary like so:

``` kotlin
@Composable
fun MyHomeScreen() {
    // Get the current SystemUiController
    val systemUiController = LocalSystemUiController.current

>     val isSystemInDarkTheme = isSystemInDarkTheme()

Reviewer: What about querying the current theme? e.g.

```
val isDarkTheme = !MaterialTheme.colors.isLight
```

Author: Done

    SideEffect {
        // Update all ofthe system bar colors to be transparent, and use
        // dark icons if we're in light theme
        systemUiController.setSystemBarsColor(
            color = Color.Transparent,
            darkIcons = !isSystemInDarkTheme
        )

        // setStatusBarsColor() and setNavigationBarsColor() also exist
    }
}
```

## System bar icon colors
The library automatically handles API level differences when running on Android devices. If we look at the example

> of status bar icons, Android only natively handles dark icons there on API 23+. This library handles this by automatically

Reviewer: ```suggestion
of status bar icons, Android only natively supports dark icons on API 23+. This library handles this by automatically
```

altering the requested color with a scrim, to maintain contrast:

![](api-scrim.png)

Similar happens on navigation bar color, which is only available on API 26+.

### Modifying scrim logic

> The scrim logic can be modified as needed by your needs:

Reviewer: ```suggestion
The scrim logic can be modified if needed:
```
